include/range/v3/view/cache1.hpp

```diff
@@ -0,0 +1,173 @@
+            {
+                return static_cast<range_value_t<Rng> &&>(std::move(parent_->cache_));
+            }
+            void next()
+            {
+                if(++current_ != ranges::end(parent_->rng_))
+                    parent_->cache_ = *current_;
+            }
+            bool equal(cursor const & that) const
+            {
+                return current_ == that.current_;
+            }
+            bool equal(sentinel const & that) const
+            {
+                return current_ == that.last_;
+            }
+            CPP_member
+            auto distance_to(cursor const & that) const -> CPP_ret(difference_type)( //
+                requires sized_sentinel_for<iterator_t<Rng>, iterator_t<Rng>>)
+            {
+                return that.current_ - current_;
+            }
+            CPP_member
+            auto distance_to(sentinel const & that) const -> CPP_ret(difference_type)( //
+                requires sized_sentinel_for<sentinel_t<Rng>, iterator_t<Rng>>)
+            {
+                return that.last_ - current_;
+            }
+        };
+
+        cursor begin_cursor()
+        {
+            auto cur = ranges::begin(rng_);
+            if(cur != ranges::end(rng_))
+                cache_ = *cur;
+            return cursor{this, std::move(cur)};
+        }
+
+        cursor end_cursor_impl(std::true_type)
+        {
+            return cursor{this, ranges::end(rng_)};
+        }
+        sentinel end_cursor_impl(std::false_type)
+        {
+            return sentinel{ranges::end(rng_)};
+        }
+        auto end_cursor()
+        {
+            return end_cursor_impl(meta::bool_<(bool)common_range<Rng>>{});
+        }
+
+    public:
+        cache1_view() = default;
+        constexpr explicit cache1_view(Rng rng)
+          : rng_{std::move(rng)}
+        {}
+        CPP_member
+        constexpr auto CPP_fun(size)()(requires sized_range<Rng>)
+        {
+            return ranges::size(rng_);
+        }
+    };
+
+#if RANGES_CXX_DEDUCTION_GUIDES >= RANGES_CXX_DEDUCTION_GUIDES_17
+    template<typename Rng>
+    cache1_view(Rng &&)->cache1_view<views::all_t<Rng>>;
+#endif
+
+    namespace views
+    {
+        struct cache1_fn
+        {
+            /// \brief Caches the most recent element within the view so that
+            /// dereferencing the view's iterator multiple types doesn't incur any
+            /// recomputation. This can be useful in adaptor pipelines that include
+            /// combinations of \c view::filter and \c view::transform, for instance.
+            /// \note \c views::cache1 is always single-pass.
+            template<typename Rng>
+            constexpr auto operator()(Rng && rng) const //
+                -> CPP_ret(cache1_view<all_t<Rng>>)(    //
+                    requires viewable_range<Rng> && input_range<Rng> &&
+                        constructible_from<range_value_t<Rng>, range_reference_t<Rng>>)
```

Comment:
Missing my `copy_constructible` from above here, and this is missing from above.